### PR TITLE
fix: scaffold copyright header

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

addressing issue #301 

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`aea hash all` and `aea packages lock --check`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: aea/cli/scaffold.py [hunk 1]
```diff
@@ -56,6 +57,50 @@
 from aea.helpers.ipfs.base import IPFSHashOnly
 
 
+COPYRIGHT_HEADER_TEMPLATE = """\
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright {year} Valory AG
```
Comment (Karrenbelt) [inline aea/cli/scaffold.py:64]: NOTE: author is still hardcoded

File: aea/cli/scaffold.py [hunk 2]
```diff
@@ -56,6 +57,50 @@
...
+def update_copyright_headers(path: Path) -> None:
+    """Update copyright headers"""
+
+    # this allows for arbitrary authors
+    year = datetime.now().year
+    top = "\n".join(COPYRIGHT_HEADER_TEMPLATE.split("\n")[:3])
+    bottom = "\n".join(COPYRIGHT_HEADER_TEMPLATE.split("\n")[-15:])
+    new_copyright_header = COPYRIGHT_HEADER_TEMPLATE.format(year=year)
+
+    for file in path.glob("**/*.py"):
```
Comment (Karrenbelt) [inline aea/cli/scaffold.py:91]: this should be `rglob`

File: aea/cli/scaffold.py (hunk 1 above)
Comment (angrybayblade) [reply aea/cli/scaffold.py:64]: Please open an issue for this we can address this on the next release

File: tests/test_cli/test_scaffold/test_connection.py [hunk 3]
```diff
@@ -91,6 +93,21 @@ def test_exit_code_equal_to_0(self):
...
+        path_pattern = f"**/{self.resource_name}/*.py"
+        for file in (Path(self.t) / self.agent_name).rglob(path_pattern):
+            content = file.read_text()
+            years_in_header = re.findall(r"Copyright.*([0-9]{4})", content)
+            if years_in_header != [str(current_year)]:
+                incorrect_files.append(file)
+            files_left_to_inspect -= 1
+        assert not "\n".join(map(str, incorrect_files))
+        assert not files_left_to_inspect
+
```
Comment (Karrenbelt) [inline tests/test_cli/test_scaffold/test_connection.py:110]: very simple tests. The `files_left_to_inspect` counter is there to ensure that the scaffolded files are in fact inspected.

File: aea/cli/scaffold.py (hunk 1 above)
Comment (DavidMinarsch) [reply aea/cli/scaffold.py:65]: It should be very easy to do this on this PR. the author can be retrieved from config and passed to this as an arg.

**Review by angrybayblade — APPROVED** [review]

**Comment by Karrenbelt** [comment]
@angrybayblade on all windows:
```
Failed tests
Exit Code	Package
1       	valory\connections\p2p_libp2p
```

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**